Views/TweetCell.m

```diff
@@ -21,4 +24,107 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     // Configure the view for the selected state
 }
 
+- (IBAction)didTapFavorite:(id)sender {
+    NSLog(@"tapped favorite");
+    if(self.tweet.favorited)
+    {
+        self.tweet.favorited = NO;
+        self.tweet.favoriteCount -= 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] unfavorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error unfavoriting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully unfavoriting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    else
+    {
+        self.tweet.favorited = YES;
+        self.tweet.favoriteCount += 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] favorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error favoriting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully favoriting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    [self refreshData];
+}
+
+- (IBAction)didTapRetweet:(id)sender {
+    NSLog(@"tapped retweet");
+    if(self.tweet.retweeted)
+    {
+        self.tweet.retweeted = NO;
+        self.tweet.retweetCount -= 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] unretweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error unretweeting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully unretweeting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    else
+    {
+        self.tweet.retweeted = YES;
+        self.tweet.retweetCount += 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] retweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error retweet tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully retweet the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    [self refreshData];
+}
+
+- (void)refreshData {
+    //MARK: setting everything inside the tweet cell
```

Comment:
It's weird to see a pragma mark inside of a method. Also, `refresh` is kind of a loaded term. This is configuring with existing data. The data isn't really fresh (recently loaded from disk or network I mean).